sensor/kubernetes/listener/watcher/crd/crdwatcher.go

```diff
@@ -73,9 +69,19 @@ func (w *crdWatcher) AddResourceToWatch(name string) error {
 }
 
 // Watch starts the CRD handler that will dispatch any events coming from k8s related to CRDs to be manage by the CRDWatcher
-func (w *crdWatcher) Watch(statusC chan<- *watcher.Status) error {
-	w.started.Store(true)
+func (w *crdWatcher) Watch(fn func(*watcher.Status)) error {
+	if w.started.Swap(true) {
+		return errors.New("Watch was already called")
+	}
+	if err := w.startHandler(); err != nil {
+		return err
+	}
```

Comment:
If we wait for the actual handler to sync instead of the informer (see my other comment), I'm not sure if this could block the handler due to the `resourceC` not being read yet.

Reply:
I can confirm that this will block if we wait for the handler to be synced.